4.gene-expression-signatures/scripts/2.run_gsea.py

```diff
@@ -85,22 +85,30 @@ def extract_weights(
             columns=dependency_df.drop(columns=["ModelID"]).columns.tolist()
         ).transpose()
         weights_df.columns = [f"{x}" for x in range(0, weights_df.shape[1])]
-    elif model_name == "betavae":
-        weights_df = weights(model, weight_data)
-        weights_df.rename(columns={0: 'genes'}, inplace=True)
-    elif model_name == "betatcvae":
-        weights_df = tc_weights(model, weight_data)
-        weights_df.rename(columns={0: 'genes'}, inplace=True)
-    elif model_name == "vanillavae":
-        weights_df = vanilla_weights(model, weight_data)
-        weights_df.rename(columns={0: 'genes'}, inplace=True)
+        weights_df = weights_df.reset_index().rename(columns={"index": "genes"})
+    elif model_name in ["betavae", "betatcvae", "vanillavae"]:
+        if model_name == "betavae":
+            weights_df = weights(model, weight_data)
+        elif model_name == "betatcvae":
+            weights_df = tc_weights(model, weight_data)
+        elif model_name == "vanillavae":
+            weights_df = vanilla_weights(model, weight_data)
```

Comment:
in the future, (note that I'm saying this without looking at the `weights()` function), I bet it would be better to write a single function (for example, `output_weights_matrix()`) that has an argument `model_name` rather than writing three separate functions. My intuition is that a lot of the functionality inside these three functions is redundant, and, that also you would avoid the if/else logic here if you could simply pass `model_name=model_name` in the single function.

Reply:
This makes sense yes